runtime/recipe/search.js

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+var assert = require('assert');
+var util = require('./util.js');
+
+// ??? rename to SearchConstraint?
```

Comment:
:man_shrugging: Either is fine with me. Slight preference for just `Search`.

Reply:
sgtm